src/MongoDbEventStoreAdapter.php

```diff
@@ -187,12 +169,16 @@ public function create(Stream $stream)
      */
     public function appendTo(StreamName $streamName, array $streamEvents)
     {
+        $this->currentStreamName = $streamName;
```

Comment:
We need to make sure that only one stream can be modified per transaction. So we should check here that either `$this->currentStreamName` is `null` or the same as `$streamName`. On transaction **commit/rollback** `$this->currentStreamName` should be reset to `null`